bnn_hmc/utils/losses.py

```diff
@@ -0,0 +1,102 @@
+# coding=utf-8
+# Copyright 2021 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Likelihood and prior functions."""
+import math
+
+import jax
+import jax.numpy as jnp
+
+from bnn_hmc.utils import tree_utils
+
+
+def make_likelihood(num_classes, temperature):
+  """Creates negative log-likelihood.
+  
+  If num_classes is None, creates regression loss. Otherwise, creates
+  cross-entropy classification loss.
+  """
+  if num_classes:
+    return make_xent_log_likelihood(num_classes, temperature)
+  else:
+    return make_gaussian_likelihood(temperature)
+
+
+def make_xent_log_likelihood(temperature):
```

Comment:
Should take in `num_classes`

Reply:
currently it infers `num_classes` from the shape of the logits, is there a reason to pass in `num_classes` as a parameter?